runtime/src/parachains.rs

```diff
@@ -729,13 +729,17 @@ impl<T: Trait> Module<T> {
 
 			// track which voters have voted already, 1 bit per authority.
 			let mut track_voters = bitvec![0; authorities.len()];
-			for (auth_index, validity_attestation) in &candidate.validity_votes {
-				let auth_index = *auth_index as usize;
+			for ((auth_index, _), validity_attestation) in candidate.validator_indices
+				.iter()
+				.enumerate()
+				.filter(|(_, bit)| *bit)
+				.zip(candidate.validity_votes.iter())
+			{
 				// protect against double-votes.
 				match validator_group.iter().find(|&(idx, _)| *idx == auth_index) {
 					None => return Err("Attesting validator not on this chain's validation duty."),
 					Some(&(idx, _)) => {
-						if track_voters.get(idx) {
+						if track_voters.get(idx).unwrap_or_default() {
```

Comment:
is this check still necessary, given that `idx` is unique in each iteration of the loop? this was not guaranteed before.